### DISCUSSION TITLE
new QA gen

### DISCUSSION THREAD

**Discussion by bogoliubon** [kilian-group/phantom-wiki#76] [Ideas]
Next steps regarding question generation 

* should we have different grammars wrt different question types? 
    In that case we can keep the current grammar untouched and have different generate functions for different grammars. 
    maybe add new cases for this function(https://github.com/albertgong1/phantom-wiki/blob/829d27fc8de03bd07d767bb3d1d377d707f9b9a9/src/phantom_wiki/facts/templates.py#L250)
    Additional grammars we can add: 
    - star join type (Yes)
    - Those from other types in the above list that we can adopt:  (do we want to? )
       Yes/No type
       comparative/superlative type 
    
* associate each type of questions with a difficulty level by defining intrinsic difficulty for different predicates. #66

**Comment by albertgong1** [comment]
Yes, during development, we should definitely have separate grammars for each question category(?)[^1]. Later on we can merge if we want questions that involve multiple question categories (e.g., "is the age of the person whose job is architect and hobby is reading comparative greater than the age of the person whose job is petroleum engineer and hobby is archeology", which would be comparative + star-join 🤯 ).

From #37 , this would be one possible grammar for star-join questions:
```
S -> T
T -> C' | 'who is the person' C' 'and' C
C -> C' | C' and C
C' -> 'whose' AN 'is' AV | 'whose' RN 'is' R | 'who has' M RN_c
M -> '<number>'
```

In my mind, the main hurdle is implementing the "and" operator in `templates.py`. Will think more about this before tomorrow's meeting.

[^1]: we already use "question type" to denote the different template sentences generated by the grammar
